be/src/olap/push_handler.cpp

```diff
@@ -239,13 +239,6 @@ Status PushHandler::_convert_v2(TabletSharedPtr cur_tablet, RowsetSharedPtr* cur
         }
         // For push load, this tablet maybe not need push data, so that the path maybe empty
         if (!path.empty()) {
-            std::unique_ptr<PushBrokerReader> reader(new (std::nothrow) PushBrokerReader());
-            if (reader == nullptr) {
-                LOG(WARNING) << "fail to create reader. tablet=" << cur_tablet->full_name();
-                res = Status::Error<MEM_ALLOC_FAILED>();
-                break;
-            }
-
             // init schema
             std::unique_ptr<Schema> schema(new (std::nothrow) Schema(tablet_schema));
```

Comment:
The variable schema is useless, just remove it and remove the parameter from PushBrokerReader's constructor.